library/core/src/num/int_log10.rs

```diff
@@ -1,76 +1,71 @@
 mod unchecked {
     // 0 < val <= u8::MAX
     pub const fn u8(val: u8) -> u32 {
-        if val >= 100 {
-            2
-        } else if val >= 10 {
-            1
-        } else {
-            0
-        }
+        let val = val as u32;
+
+        // For better performance, avoid branches by assembling the solution
+        // in the bits above the low 8 bits.
+
+        // Adding c1 to val gives 10 in the top bits for val < 10, 11 for val >= 10
+        let c1 = 0b11_00000000 - 10; // 758
+        // Adding c2 to val gives 01 in the top bits for val < 100, 10 for val >= 100
+        let c2 = 0b10_00000000 - 100; // 412
```

Comment:
It's a really minor thing, but shouldn't these be constants?

Reply:
OK, done